v1.0 rc

blackfalcon opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is, and anything that describes the before/after state of the project._

This is the summary PR for releasing the v1.0 version of auro-toast. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

cynthiaricomendoza-alaskaair committed:
Newly generated repo commit

This commit adds all the new resources created
from the Auro WC Generator.

Changes to be committed:
modified:   .github/CONTRIBUTING.md
modified:   .github/ISSUE_TEMPLATE/blueprint.md
new file:   .github/ISSUE_TEMPLATE/bug_report.yml
new file:   .github/ISSUE_TEMPLATE/feature_request.yml
new file:   .github/ISSUE_TEMPLATE/general-support.yml
modified:   .github/ISSUE_TEMPLATE/publish_component.md
modified:   .github/settings.yml
new file:   .github/workflows/codeql.yml
new file:   .github/workflows/testPublish.yml
modified:   .gitignore
new file:   .husky/_/husky.sh
new file:   .husky/commit-msg
new file:   .husky/pre-commit
modified:   .npmignore
modified:   .stylelintrc
modified:   NOTICE
modified:   README.md
new file:   apiExamples/basic.html
new file:   commitlint.config.js
new file:   demo/apiExamples.md
new file:   demo/demo.md
new file:   demo/index.html
new file:   docs/api.md
new file:   docs/partials/apiExamples.md
new file:   docs/partials/demo.md
new file:   docs/partials/description.md
new file:   docs/partials/readmeAddlInfo.md
new file:   docs/partials/useCases.md
new file:   docs/upstream.md
new file:   index.js
new file:   package-lock.json
new file:   package.json
new file:   packageScripts/postinstall.mjs
new file:   rollup.config.mjs
new file:   scripts/generateDocs.js
new file:   scripts/postCss.js
new file:   scripts/pre-commit.mjs
new file:   scripts/staticStyles-template.js
new file:   src/auro-toast.js
new file:   src/style.scss
new file:   test/auro-toast.test.js
new file:   tsconfig.json
new file:   web-test-runner.config.mjs

cynthiaricomendoza-alaskaair committed:
build(CODEOWNERS): add innersource reviewers #3

Changes to be committed:
modified:   .github/CODEOWNERS

cynthiaricomendoza-alaskaair committed:
feat(ux/ui): hide toast after five seconds #13

cynthiaricomendoza-alaskaair committed:
feat(ux/ui): build of component #5

DZwell committed:
chore(generateDocs): remove superfluous code #7

cynthiaricomendoza-alaskaair committed:
feat: support type property #8

cynthiaricomendoza-alaskaair committed:
feat: add close button #10

cynthiaricomendoza-alaskaair committed:
feat: placement of toast on UI #14

cynthiaricomendoza-alaskaair committed:
feat(ux/ui): implement toaster #19

This commit completely implement the toaster.js feature
that will manage the experience of multiple toasts.

Changes to be committed:
modified:   README.md
modified:   apiExamples/basic.html
modified:   apiExamples/error.html
new file:   apiExamples/multipleToasts.html
modified:   apiExamples/success.html
renamed:    demo/toggleToast.js -> demo/addToast.js
modified:   demo/apiExamples.md
modified:   demo/demo.md
modified:   demo/index.html
modified:   docs/api.md
modified:   docs/partials/demo.md
modified:   docs/partials/description.md
modified:   docs/partials/useCases.md
modified:   package-lock.json
modified:   package.json
modified:   src/auro-toast.js
new file:   src/auro-toaster.js
new file:   src/style-toaster.scss
modified:   src/style.scss
modified:   test/auro-toast.test.js

cynthiaricomendoza-alaskaair committed:
feat(ux/ui): support noIcon property #9

cynthiaricomendoza-alaskaair committed:
feat: do not auto-dismiss error toast #26

cynthiaricomendoza-alaskaair committed:
perf(mobile): allow greater touch surface #22

This commit will allow users to tab anywhere inside
the toast when dismissing.

Changes to be committed:
modified:   src/auro-toast.js
modified:   src/style-toaster.scss
modified:   src/style.scss

cynthiaricomendoza-alaskaair committed:
chore(deps): remove unused dependency

this commit will remove the dependency @open-wc/testing-karma, since we don't use it anymore

Changes to be committed:
modified:   package-lock.json
modified:   package.json

cynthiaricomendoza-alaskaair committed:
docs(api): add property examples

this commit adds indiviual examples for the properties the toast supports

Changes to be committed:
new file:   apiExamples/noIcon.html
new file:   apiExamples/variant.html
new file:   apiExamples/visible.html
modified:   demo/apiExamples.md
modified:   docs/partials/apiExamples.md

cynthiaricomendoza-alaskaair committed:
docs(readme): cleanup dynamic toasts & custom element

this commit adds some code to better the example for the dynamic toasts section and cleans up the custom element section so it works and shows correctly

Changes to be committed:
modified:   demo/demo.md
modified:   demo/index.html
modified:   docs/partials/demo.md

cynthiaricomendoza-alaskaair committed:
refactor(style): cleanup toast classes

this commit takes care of the following:
removes the usage of an ID on a element in the toast and replaces it with a class
removes styles for button element since there is a class that wraps around the button and can take care of the styles there

Changes to be committed:
modified:   src/auro-toast.js
modified:   src/style.scss
modified:   test/auro-toast.test.js

cynthiaricomendoza-alaskaair committed:
perf(demo): create api.html file

this commit creates the api.html file to help review the apiExmaples.md file

Changes to be committed:
new file:   demo/api.html

cynthiaricomendoza-alaskaair committed:
perf(screenreader): notify when toast is visible

this commit adds aria-live="polite" to the auro-toast, so the screen reader captures the toast and reads that it is visible

Changes to be committed:
modified:   src/auro-toast.js

cynthiaricomendoza-alaskaair committed:
docs(api): add missing scripts

this commit adds scripts needed to perform showing the toast when clicking on the buttons

Changes to be committed:
modified:   demo/api.html
modified:   demo/apiExamples.md
modified:   demo/index.html
modified:   docs/api.md
modified:   src/auro-toast.js

cynthiaricomendoza-alaskaair committed:
docs(readme): add script for toaster

this commit adds the script for toaster so the mutliple toast section in the demo works correctly.

Changes to be committed:
modified:   demo/index.html

cynthiaricomendoza-alaskaair commented on .github/settings.yml:
```diff
@@ -3,13 +3,13 @@ repository:
 
   # A short description of the repository that will show up on GitHub
-  description: Placeholder repo
+  description: Custom element that ....
```
```suggestion
  description: Custom element that illustrates a quick message after an action is complete
```

blackfalcon committed:
docs(*): update demo and example docs

Changes to be committed:
modified:   README.md
modified:   apiExamples/basic.html
modified:   apiExamples/error.html
modified:   apiExamples/multipleToasts.html
modified:   apiExamples/variant.html
modified:   apiExamples/visible.html
modified:   demo/apiExamples.md
modified:   demo/demo.md
modified:   docs/partials/apiExamples.md
modified:   docs/partials/demo.md
modified:   docs/partials/description.md
modified:   docs/partials/useCases.md

blackfalcon committed:
docs(jsdoc): update function description

Changes to be committed:
modified:   src/auro-toast.js

blackfalcon committed:
fix(lit): update template to import from correct ref

Changes to be committed:
modified:   scripts/staticStyles-template.js

blackfalcon committed:
chore(settings): update element description

Changes to be committed:
modified:   .github/settings.yml

blackfalcon committed:
docs(issues): delete deprecated versions of issue templates

Changes to be committed:
deleted:    .github/ISSUE_TEMPLATE/blueprint.md
deleted:    .github/ISSUE_TEMPLATE/dev-story.md
deleted:    .github/ISSUE_TEMPLATE/feature_request.md
deleted:    .github/ISSUE_TEMPLATE/general-support.md
deleted:    .github/ISSUE_TEMPLATE/publish_component.md
deleted:    .github/ISSUE_TEMPLATE/release_candidate.md

cynthiaricomendoza-alaskaair reviewed (dismissed):
lgtm - thanks for making those editorial changes!

blackfalcon commented on scripts/generateDocs.js:
````diff
@@ -0,0 +1,200 @@
+  result = result.replace(/>(\r\n|\r|\n){2,}/g, '>\r\n'); // Remove empty lines directly after a closing html tag.
+  result = result.replace(/>(\r\n|\r|\n)```/g, '>\r\n\r\n```'); // Ensure an empty line before code samples.
+  result = result.replace(/>(\r\n|\r|\n){2,}```(\r\n|\r|\n)/g, '>\r\n```\r\n'); // Ensure no empty lines before close of code sample.
+  result = result.replace(/([^(\r\n|\r|\n)])(\r\n|\r|\n)+#/g, "$1\r\n\r\n#"); // Ensure empty line before header sections.
````
The following PR is being worked on to address this issue 

- https://github.com/AlaskaAirlines/WC-Generator/pull/410

blackfalcon commented:
Chrome mobile experience 

<img width="415" alt="Screen Shot 2023-07-10 at 3 21 02 PM" src="https://github.com/AlaskaAirlines/auro-toast/assets/181089/8fbcefdc-eb98-4a5e-8257-06ea05e8451f">

Safari 

<img width="674" alt="Screen Shot 2023-07-10 at 3 22 32 PM" src="https://github.com/AlaskaAirlines/auro-toast/assets/181089/2341d5e0-cd8d-4c42-ba40-444abfce3ac0">

Firefox

<img width="738" alt="Screen Shot 2023-07-10 at 3 23 18 PM" src="https://github.com/AlaskaAirlines/auro-toast/assets/181089/fb3dea90-1523-4009-be1f-a07a4f4f56e2">

blackfalcon commented:
Lighthouse A11Y score. Issue is NOT related to the component, has to do with the coloring of the code examples. 

<img width="543" alt="Screen Shot 2023-07-10 at 3 24 55 PM" src="https://github.com/AlaskaAirlines/auro-toast/assets/181089/212047ad-0bce-4423-b7f4-ec975a915ef1">

jordanjones243 commented on .github/workflows/codeql.yml:
```diff
@@ -0,0 +1,58 @@
+name: "CodeQL"
```
Missing `autoAssign.yml`

blackfalcon replied:
Added to the repo.

jordanjones243 commented on demo/addToast.js:
```diff
@@ -0,0 +1,12 @@
+
+// ---------------------------------------------------------------------
+
+showToast = (toastID) => {
```
This should be located under `apiExamples/` and should be ran under a file that runs all javascript examples for each respective page, like so: https://github.com/AlaskaAirlines/auro-combobox/blob/main/demo/index.js

blackfalcon replied:
@jordanjones243 please consult with @cynthiaricomendoza-alaskaair to assist with this update. This follows the new example.js model that I am not all that failure with.

cynthiaricomendoza-alaskaair replied:
met with Jordan yesterday and pushed up changes for this!

jordanjones243 commented on rollup.config.mjs:
```diff
@@ -0,0 +1,24 @@
+  ]
+};
+
+export default [modernConfig];
```
Need to update rollup.config: https://github.com/AlaskaAirlines/auro-combobox/blob/main/rollup.config.mjs

blackfalcon replied:
@jordanjones243 again, please follow up with @cynthiaricomendoza-alaskaair on this. 

Also, this model is not part of the generator. I was unaware that this was a pattern that we are actively doing with all components. 

@Patrick-Daly-AA there should be an issue created with the WC-Generator to add this functionality. Please work with @jordanjones243 to define that specification.

cynthiaricomendoza-alaskaair replied:
met with Jordan yesterday and pushed up changes for this!

jordanjones243 reviewed (changes requested):
Need to run a build to update docs, an event type was changed.

blackfalcon committed:
ci(config): add auto assign issue yml

Changes to be committed:
new file:   .github/workflows/autoAssign.yml

cynthiaricomendoza-alaskaair committed:
chore: use new example.js model

this commit is to follow the new js model

Changes to be committed:
modified:   .gitignore
renamed:    demo/addToast.js -> apiExamples/showToast.js
modified:   demo/api.html
new file:   demo/api.js
modified:   demo/apiExamples.md
modified:   demo/index.html
new file:   demo/index.js
modified:   docs/api.md
modified:   package.json
modified:   rollup.config.mjs
new file:   scripts/prepExampleFiles.js

jordanjones243 commented on demo/index.html:
```diff
@@ -43,13 +43,16 @@
     <script type="module" src="../src/auro-toaster.js"></script>
-    <script src="./addToast.js"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-button@latest/dist/auro-button__bundled.js" type="module"></script>
+    <script type="module" src="./index.min.js"></script>
```
type="module" needs to be removed from this line and the one below

cynthiaricomendoza-alaskaair replied:
done! all pushed up

jordanjones243 commented on demo/index.js:
```diff
@@ -0,0 +1,16 @@
+  initCount = initCount || 0;
+
+  try {
+    showToast();
```
Comment this line out as it is no longer needed and leave comment above this line saying "add new javascript function calls to test examples on index page"

Do the same thing for the api page

cynthiaricomendoza-alaskaair replied:
i added a comment, but kept showToast() since that is needed in order for the button to work.

jordanjones243 commented on docs/partials/apiExamples.md:
````diff
@@ -0,0 +1,64 @@
+  <span slot="trigger">See code</span>
+
+```js
+// Example function
````
This needs to be updated to auto generate the script with `apiExamples/showToast.js` as you see on line 13 and 14. There needs to be documentation explaining that every button is using this file.

cynthiaricomendoza-alaskaair replied:
update in this commit: https://github.com/AlaskaAirlines/auro-toast/pull/27/commits/4dd55edb65c0ac21070401e7ce8c9c5f295cad6b

cynthiaricomendoza-alaskaair committed:
docs(api): add showToast() in all examples

this commit adds the auto-generated script for the showToast function in all the examples so users are aware this is used for all buttons on this page

Changes to be committed:
modified:   apiExamples/showToast.js
modified:   demo/apiExamples.md
modified:   docs/partials/apiExamples.md

blackfalcon commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-toast/v/1.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-toast/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
